src/index.js

```diff
@@ -0,0 +1,105 @@
+import "./style.css";
+import { getSingeMeal, getAllMeal } from "./modules/getMeals.js";
+import { displayInHtml, cardContainer } from "./modules/displayMeals.js";
+import generatePopupMarkup from "./modules/generatePopupMarkup.js";
+import {
+  getComments,
+  insertCommentMarkup,
+  postComment,
+} from "./modules/commentsHandler.js";
+import { postLikes, getLikes } from "./modules/likesHandler.js";
+import navigationHandler from "./modules/navigationHandler.js";
+
+const popupHook = document.querySelector(".popup-hook");
+const itemContainer = document.querySelector(".cards");
+const itemCount = document.querySelector(".item-count");
+
+(async () => {
+  const meals = await getAllMeal();
+  const likes = await getLikes();
+  itemCount.textContent = meals.length;
```

Comment:
- Kindly make sure to follow the guidelines in the card `Add all items counter on the Homepage - Student A` in this [issues](https://github.com/serengia/capstone-project--consuming-API/issues/6): 
   - [ ] You need to implement the counting logic in a separate module.
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

Reply:
Hello @gemmen29 . Thank you for your feedback. I have been struggling for many hours without success to make the changes you requested work. Also, after several attempts, I still do not clearly understand the process I need to follow to implement the changes. 
### My concern
Besides the above-highlighted issues I am facing, I feel the requested changes will go against the DRY and KISS principles. Why do I think so? In my case, I am already passing the comments array (from which I can get the length) in my markup. Having a separate function to calculate and insert the comments count for every item will be unnecessary repetition besides making it harder to read my code.
Kindly advice. I feel stuck.